Sources/Commands/Workspace.swift

```diff
@@ -216,6 +236,54 @@ public class Workspace {
         }
     }
 
+    /// Find the managed dependencies for the given package name.
+    ///
+    /// It is possible that the provided package name is correct but
+    /// dependency is not yet fetched. This method will throw an error
+    /// in that case.
+    ///
+    /// - Throws: WorkspaceOperationError
+    func findDependency(package: String) throws -> ManagedDependency {
+        // Load the external manifests.
+        let (_, manifests) = try loadDependencyManifests()
+        // Find the dependency.
+        guard let url = manifests.first(where: { $0.name == package })?.url,
+              let dependency = dependencyMap[RepositorySpecifier(url: url)] else {
+            throw WorkspaceOperationError.dependencyNotFound(package)
+        }
+        return dependency
+    }
+
+    /// Puts a dependency in edit mode creating a checkout in edits directory.
+    func edit(dependency: ManagedDependency, at revision: Revision) throws {
+        // Ensure that the dependency is not already in edit mode.
+        guard !dependency.isEdited else {
+            throw WorkspaceOperationError.dependencyAlreadyInEditMode
+        }
+        let repository = dependency.repository
+
+        // Compute new path for the dependency.
+        // FIXME: Users will be interacting with this directory but the name still
+        // has to be unique. Maybe generate a prettified but unique name.
+        let path = editsPath.appending(component: repository.fileSystemIdentifier)
+
+        // FIXME: If we are editing a dependency then we already have it fetched.
+        // Should we still wait for the lookup?
+        let handle = checkoutManager.lookup(repository: repository)
+
+        try handle.cloneCheckout(to: path, editable: true)
+        let workingRepo = try checkoutManager.provider.openCheckout(at: path)
+        try workingRepo.checkout(revision: revision)
+
+        // Change its stated to edited.
+        // FIXME: We lose the previous state, do we care?
```

Comment:
What part do we lose? Wouldn't it be clearer to just create a `var editableDependency = dependency; editableEdependency.isEdited = true`?